evmrpc/simulate.go

```diff
@@ -503,7 +503,7 @@ func (b *Backend) getHeader(blockNumber *big.Int) *ethtypes.Header {
 	header := &ethtypes.Header{
 		Difficulty:    common.Big0,
 		Number:        blockNumber,
-		BaseFee:       nil,
+		BaseFee:       b.keeper.GetNextBaseFeePerGas(b.ctxProvider(blockNumber.Int64() - 1)).TruncateInt().BigInt(),
```

Comment:
let's make this upgrade height dependent too (i.e. still use nil if blockNumber is below 6.2.0 upgrade height). See the other PR (the simplify one) for code snippet